<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>800</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Elevator Simulation</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QWidget" name="layoutWidget">
       <layout class="QVBoxLayout" name="leftLayout">
        <item>
         <widget class="QGroupBox" name="simulationControlsGroupBox">
          <property name="title">
           <string>Simulation Controls</string>
          </property>
          <layout class="QGridLayout" name="gridLayout">
           <item row="0" column="0">
            <widget class="QLabel" name="floorsLabel">
             <property name="text">
              <string>Floors:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QSpinBox" name="floorsSpinBox">
             <property name="minimum">
              <number>2</number>
             </property>
             <property name="maximum">
              <number>20</number>
             </property>
             <property name="value">
              <number>5</number>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QLabel" name="elevatorsLabel">
             <property name="text">
              <string>Elevators:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="3">
            <widget class="QSpinBox" name="elevatorsSpinBox">
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>10</number>
             </property>
             <property name="value">
              <number>2</number>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="passengersLabel">
             <property name="text">
              <string>Passengers:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QSpinBox" name="passengersSpinBox">
             <property name="minimum">
              <number>0</number>
             </property>
             <property name="maximum">
              <number>20</number>
             </property>
             <property name="value">
              <number>3</number>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <widget class="QLabel" name="strategyLabel">
             <property name="text">
              <string>Strategy:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="3">
            <widget class="QComboBox" name="strategyComboBox">
             <item>
              <property name="text">
               <string>Nearest Elevator</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Least Busy</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Direction-based</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="2" column="0" colspan="4">
            <layout class="QHBoxLayout" name="horizontalLayout">
             <item>
              <widget class="QPushButton" name="startButton">
               <property name="text">
                <string>Start</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="pauseButton">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="text">
                <string>Pause</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="resumeButton">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="text">
                <string>Resume</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="stepButton">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="text">
                <string>Step</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="stopButton">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="text">
                <string>Stop</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="resetButton">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="text">
                <string>Reset</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="3" column="0" colspan="2">
            <widget class="QLabel" name="timeStepLabel">
             <property name="text">
              <string>Time Step: 0</string>
             </property>
            </widget>
           </item>
           <item row="3" column="2" colspan="2">
            <widget class="QLabel" name="stateLabel">
             <property name="text">
              <string>State: Setup</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="passengerEventsGroupBox">
          <property name="title">
           <string>Passenger Events</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_2">
           <item row="0" column="0">
            <widget class="QLabel" name="passengerIdLabel">
             <property name="text">
              <string>Passenger ID:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QSpinBox" name="passengerIdSpinBox">
             <property name="minimum">
              <number>0</number>
             </property>
             <property name="maximum">
              <number>19</number>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QLabel" name="eventTypeLabel">
             <property name="text">
              <string>Event Type:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="3">
            <widget class="QComboBox" name="eventTypeComboBox">
             <item>
              <property name="text">
               <string>Request Elevator</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Enter Elevator</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Press Floor Button</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Press Help Button</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Press Door Open</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Press Door Close</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Exit Elevator</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="timeStepEventLabel">
             <property name="text">
              <string>Time Step:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QSpinBox" name="timeStepEventSpinBox">
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>100</number>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <widget class="QLabel" name="floorLabel">
             <property name="text">
              <string>Floor:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="3">
            <widget class="QSpinBox" name="floorSpinBox">
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>20</number>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="directionLabel">
             <property name="text">
              <string>Direction:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QComboBox" name="directionComboBox">
             <item>
              <property name="text">
               <string>UP</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>DOWN</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="2" column="2">
            <widget class="QLabel" name="targetFloorLabel">
             <property name="text">
              <string>Target Floor:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="3">
            <widget class="QSpinBox" name="targetFloorSpinBox">
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>20</number>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="elevatorIdLabel">
             <property name="text">
              <string>Elevator ID:</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QSpinBox" name="elevatorIdSpinBox">
             <property name="minimum">
              <number>0</number>
             </property>
             <property name="maximum">
              <number>9</number>
             </property>
            </widget>
           </item>
           <item row="3" column="2" colspan="2">
            <widget class="QPushButton" name="addPassengerEventButton">
             <property name="text">
              <string>Add Event</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="safetyEventsGroupBox">
          <property name="title">
           <string>Safety Events</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_3">
           <item row="0" column="0">
            <widget class="QLabel" name="safetyEventTypeLabel">
             <property name="text">
              <string>Event Type:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QComboBox" name="safetyEventTypeComboBox">
             <item>
              <property name="text">
               <string>Help Button</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Door Obstacle</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Fire Alarm</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Overload</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Power Out</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="safetyTimeStepLabel">
             <property name="text">
              <string>Time Step:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QSpinBox" name="safetyTimeStepSpinBox">
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>100</number>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="safetyElevatorIdLabel">
             <property name="text">
              <string>Elevator ID:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QSpinBox" name="safetyElevatorIdSpinBox">
             <property name="minimum">
              <number>-1</number>
             </property>
             <property name="maximum">
              <number>9</number>
             </property>
             <property name="value">
              <number>-1</number>
             </property>
            </widget>
           </item>
           <item row="3" column="0" colspan="2">
            <widget class="QPushButton" name="addSafetyEventButton">
             <property name="text">
              <string>Add Safety Event</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="layoutWidget1">
       <layout class="QVBoxLayout" name="rightLayout">
        <item>
         <widget class="QGroupBox" name="buildingDisplayGroupBox">
          <property name="title">
           <string>Building Display</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <widget class="QScrollArea" name="buildingScrollArea">
             <property name="widgetResizable">
              <bool>true</bool>
             </property>
             <widget class="QWidget" name="buildingScrollAreaContents">
              <property name="geometry">
               <rect>
                <x>0</x>
                <y>0</y>
                <width>454</width>
                <height>300</height>
               </rect>
              </property>
              <layout class="QVBoxLayout" name="buildingLayout"/>
             </widget>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="logGroupBox">
          <property name="title">
           <string>Log Console</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QTextEdit" name="logTextEdit">
             <property name="readOnly">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1000</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui> 